cli/internal/core/engine.go

```diff
@@ -153,10 +153,11 @@ func (e *Engine) generateTaskGraph(pkgs []string, taskNames []string, tasksOnly
 		}
 
 		visited.Add(taskID)
-		deps := task.Deps
 
+		// Filter down the tasks if there's a filter in place
+		// https: //turbo.build/repo/docs/reference/command-line-reference#--only
 		if tasksOnly {
-			deps = deps.Filter(func(d interface{}) bool {
+			task.Deps = task.Deps.Filter(func(d interface{}) bool {
```

Comment:
Is there a reason for the change to update the stored Task object instead of a local var?

Reply:
I had the same question the other way around and din't know the answer. That's why i made the change. `topoDeps` gets updated on the `Task` object:

https://github.com/vercel/turbo/blob/652408b2167453c5f325ef4d0b7ffd351a386c8a/cli/internal/core/engine.go#L160

I could change _that_ to be a local variable instead for consistency? (so we're not questioning the choice each time we read the code). I am *mostly* sure that the difference is not important, but I could be wrong I guess. Can you think of any reason not to update the Task object in both cases?

Reply:
So, both directions are safe here (I just checked), but we should be using local vars for the mutation. The `Task` object should map to the "at rest" state, and the filtered state is consumed just by the engine.

We'll care more about this when we have the graph state provided by the daemon (immutable, a picture of the world) and we're referring to it in order to identify what to do.